2_task/arrayOperation.js

```diff
@@ -0,0 +1,154 @@
+//function isArray
+function isArrayByInstance(arr) {
+  return arr instanceof Array;
+}
+
+function isArrayByPrototype(arr) {
+  return Object.prototype.toString.call(arr) === '[object Array]';
+}
+
+function isArrayByDuckTyping(arr) {
+  if (arr.splice) {
+    return true;
+  }
+  return false;
+}
+//function range
+function range(start, end, multiply) {
```

Comment:
range(40, 20) should return [] but range(40, 20, -1) should return [40, 39, 38, 37, 36, 35, 34, 33, 32, 31, 30, 29, 28, 27, 26, 25, 24, 23, 22, 21]
same with range(-20, -40)